python/glide-shared/glide_shared/config.py

```diff
@@ -275,13 +305,22 @@ def _create_a_protobuf_conn_request(
         if self.connection_timeout:
             request.connection_timeout = self.connection_timeout
 
-        if self.tls_config and self.tls_config.use_insecure_tls:
-            if request.tls_mode == TlsMode.SecureTls:
-                request.tls_mode = TlsMode.InsecureTls
-            elif request.tls_mode == TlsMode.NoTls:
-                raise ConfigurationError(
-                    "use_insecure_tls cannot be enabled when use_tls is disabled."
-                )
+        if self.tls_config:
+            if self.tls_config.use_insecure_tls:
```

Comment:
I have the same confusion about this as in node.js:
https://github.com/valkey-io/valkey-glide/pull/4911/files#r2478980397